staging/src/k8s.io/api/events/v1/types.go

```diff
@@ -27,7 +27,7 @@ import (
 // Event is a report of an event somewhere in the cluster. It generally denotes some state change in the system.
 type Event struct {
 	metav1.TypeMeta `json:",inline"`
-	// +optional
+
```

Comment:
Given you touch this file, if you create objects with the new API, more fields are actually required
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/events.go#L41

xref https://github.com/kubernetes/kubernetes/issues/94748

Can you please fix that in the same PR too?

Reply:
Also, please fix the v1beta1 api group.

Reply:
@wojtek-t regarding the v1beta1.. I made the changes locally..
But regarding the events `events.k8s.io` one, I have a doubt. So, by fixing, do you mean that the strict validations should not be necessary even in case of `events.k8s.io/Event` and hence, I should edit those validation checks?
OR
Do you mean that I should just add "Required" comments in `staging/src/k8s.io/api/events/v1/types.go` under `type Event` so as to convey that the corresponding fields are actually required by the Validator function?

Reply:
> Do you mean that I should just add "Required" comments under type Event (inside staging/src/k8s.io/api/events/v1/types.go) so as to convey that the corresponding fields are actually required by the Validator function?

Yes - this one.

Reply:
Alright. On it!

Reply:
> > noob question. What do you mean by regenerating the autogenerated files? How to do it? (No need to explain it if there's any existing documentation on this somewhere in the repo)
> 
> ohh sorry
> `make update`
> is what should do the job for you

Faced an error with this previously and spend quite a lot of time on resolving it but couldn't work it out.
(Go version: 1.15.2)
(`uname -a`:   `Linux luci4 5.4.0-47-generic #51~18.04.1-Ubuntu SMP Sat Sep 5 14:35:50 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`)

Sorry for the huge incoming log but **the error is in the end (seemingly)**

```
► make update
# unicode/utf8
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# math/bits
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# internal/race
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# internal/goversion
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# internal/nettrace
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# runtime/internal/sys
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# sync/atomic
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
        predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
        include directory; can be set multiple times
  -S    print assembly and machine code
  -V    print version and exit
  -debug
        dump instructions as they are parsed
  -dynlink
        support references to Go symbols defined in other shared libraries
  -e    no limit on number of errors reported
  -gensymabis
        write symbol ABI information to output file, don't assemble
  -newobj
        use new object file format
  -o string
        output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
        generate code that can be linked into a shared library
  -trimpath string
        remove prefix from recorded source file paths
# internal/cpu
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
        predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
        include directory; can be set multiple times
  -S    print assembly and machine code
  -V    print version and exit
  -debug
        dump instructions as they are parsed
  -dynlink
        support references to Go symbols defined in other shared libraries
  -e    no limit on number of errors reported
  -gensymabis
        write symbol ABI information to output file, don't assemble
  -newobj
        use new object file format
  -o string
        output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
        generate code that can be linked into a shared library
  -trimpath string
        remove prefix from recorded source file paths
# unicode
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# runtime/internal/atomic
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
        predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
        include directory; can be set multiple times
  -S    print assembly and machine code
  -V    print version and exit
  -debug
        dump instructions as they are parsed
  -dynlink
        support references to Go symbols defined in other shared libraries
  -e    no limit on number of errors reported
  -gensymabis
        write symbol ABI information to output file, don't assemble
  -newobj
        use new object file format
  -o string
        output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
        generate code that can be linked into a shared library
  -trimpath string
        remove prefix from recorded source file paths
# runtime/cgo
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
        predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
        include directory; can be set multiple times
  -S    print assembly and machine code
  -V    print version and exit
  -debug
        dump instructions as they are parsed
  -dynlink
        support references to Go symbols defined in other shared libraries
  -e    no limit on number of errors reported
  -gensymabis
        write symbol ABI information to output file, don't assemble
  -newobj
        use new object file format
  -o string
        output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
        generate code that can be linked into a shared library
  -trimpath string
        remove prefix from recorded source file paths
+++ [0925 00:25:09] Building go targets for linux/amd64:
    ./vendor/k8s.io/code-generator/cmd/prerelease-lifecycle-gen
# runtime/internal/sys
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# unicode/utf8
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# math/bits
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# unicode
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# internal/race
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# internal/goversion
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# internal/nettrace
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# runtime/internal/atomic
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
        predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
        include directory; can be set multiple times
  -S    print assembly and machine code
  -V    print version and exit
  -debug
        dump instructions as they are parsed
  -dynlink
        support references to Go symbols defined in other shared libraries
  -e    no limit on number of errors reported
  -gensymabis
        write symbol ABI information to output file, don't assemble
  -newobj
        use new object file format
  -o string
        output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
        generate code that can be linked into a shared library
  -trimpath string
        remove prefix from recorded source file paths
# unicode/utf16
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# internal/cpu
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
        predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
        include directory; can be set multiple times
  -S    print assembly and machine code
  -V    print version and exit
  -debug
        dump instructions as they are parsed
  -dynlink
        support references to Go symbols defined in other shared libraries
  -e    no limit on number of errors reported
  -gensymabis
        write symbol ABI information to output file, don't assemble
  -newobj
        use new object file format
  -o string
        output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
        generate code that can be linked into a shared library
  -trimpath string
        remove prefix from recorded source file paths
# encoding
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# k8s.io/kubernetes/vendor/golang.org/x/mod/semver
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# k8s.io/kubernetes/vendor/golang.org/x/xerrors/internal
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# sync/atomic
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
        predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
        include directory; can be set multiple times
  -S    print assembly and machine code
  -V    print version and exit
  -debug
        dump instructions as they are parsed
  -dynlink
        support references to Go symbols defined in other shared libraries
  -e    no limit on number of errors reported
  -gensymabis
        write symbol ABI information to output file, don't assemble
  -newobj
        use new object file format
  -o string
        output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
        generate code that can be linked into a shared library
  -trimpath string
        remove prefix from recorded source file paths
# k8s.io/kubernetes/vendor/github.com/go-logr/logr
compile: version "go1.14.6" does not match go tool version "go1.15.2"
# runtime/cgo
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
        predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
        include directory; can be set multiple times
  -S    print assembly and machine code
  -V    print version and exit
  -debug
        dump instructions as they are parsed
  -dynlink
        support references to Go symbols defined in other shared libraries
  -e    no limit on number of errors reported
  -gensymabis
        write symbol ABI information to output file, don't assemble
  -newobj
        use new object file format
  -o string
        output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
        generate code that can be linked into a shared library
  -trimpath string
        remove prefix from recorded source file paths
!!! [0925 00:25:09] Call tree:
!!! [0925 00:25:09]  1: /home/yash1300/Projects/GoProjects/github.com/yashvardhan-kukreja/kubernetes/hack/lib/golang.sh:719 kube::golang::build_some_binaries(...)
!!! [0925 00:25:09]  2: /home/yash1300/Projects/GoProjects/github.com/yashvardhan-kukreja/kubernetes/hack/lib/golang.sh:863 kube::golang::build_binaries_for_platform(...)
!!! [0925 00:25:09]  3: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0925 00:25:09] Call tree:
!!! [0925 00:25:09]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0925 00:25:09] Call tree:
!!! [0925 00:25:09]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
Makefile.generated_files:195: recipe for target '_output/bin/prerelease-lifecycle-gen' failed
make[1]: *** [_output/bin/prerelease-lifecycle-gen] Error 1
Makefile:546: recipe for target 'generated_files' failed
make: *** [generated_files] Error 2
```